consensus/state_sync/state_sync.go

```diff
@@ -0,0 +1,132 @@
+package state_sync
+
+import (
+	"fmt"
+	"log"
+
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+const (
+	DefaultLogPrefix    = "NODE"
+	stateSyncModuleName = "stateSyncModule"
+)
+
+type SyncMode string
+
+const (
+	Sync      SyncMode = "sync"
+	Synched   SyncMode = "synched"
+	Pacemaker SyncMode = "pacemaker"
+	Server    SyncMode = "server"
+)
+
+type StateSyncModule interface {
+	modules.Module
+	StateSyncServerModule
+
+	// Handle a metadata response from a peer so this node can update its local view of the state
+	// sync metadata available from its peers
+	HandleStateSyncMetadataResponse(*typesCons.StateSyncMetadataResponse) error
+
+	// Handle a block response from a peer so this node can update apply it to its local state
+	// and catch up to the global world state
+	HandleGetBlockResponse(*typesCons.GetBlockResponse) error
+
+	IsServerModEnabled() bool
+	EnableServerMode()
+	SetLogPrefix(string)
+}
+
+var (
+	_ modules.Module        = &stateSync{}
+	_ StateSyncModule       = &stateSync{}
+	_ StateSyncServerModule = &stateSync{}
+)
+
+type stateSync struct {
+	bus modules.Bus
+
+	currentMode SyncMode
+	serverMode  bool
+
+	logPrefix string
+}
+
+func CreateStateSync(bus modules.Bus) (modules.Module, error) {
+	var m stateSync
+	return m.Create(bus)
+}
+
+func (*stateSync) Create(bus modules.Bus) (modules.Module, error) {
+	m := &stateSync{}
+	bus.RegisterModule(m)
+
+	// when node is starting, it is in sync mode, as it might need to bootstrap to the latest state
+	m.currentMode = Sync
+
+	m.serverMode = false
+
+	return m, nil
+}
+
+func (m *stateSync) Start() error {
+	return nil
+}
+
+func (m *stateSync) Stop() error {
+	return nil
+}
+
+func (m *stateSync) SetBus(pocketBus modules.Bus) {
+	m.bus = pocketBus
+}
+
+func (m *stateSync) GetBus() modules.Bus {
+	if m.bus == nil {
+		log.Fatalf("PocketBus is not initialized")
+	}
+	return m.bus
+}
+
+func (m *stateSync) GetModuleName() string {
+	return stateSyncModuleName
+}
+
+func (m *stateSync) IsServerModEnabled() bool {
+	return m.serverMode
+}
+
+func (m *stateSync) SetLogPrefix(logPrefix string) {
+	m.logPrefix = logPrefix
+}
+
+func (m *stateSync) EnableServerMode() {
+	m.currentMode = Server
+	m.serverMode = true
+}
+
+// TODO implement
+// Placeholder function
+func (m *stateSync) HandleGetBlockResponse(blockRes *typesCons.GetBlockResponse) error {
+	m.nodeLog(fmt.Sprintf("Received get block response: %s", blockRes.Block.String()))
+	return nil
+}
+
+// TODO implement
+// Placeholder function
+func (m *stateSync) HandleStateSyncMetadataResponse(metaDataRes *typesCons.StateSyncMetadataResponse) error {
+	m.nodeLog(fmt.Sprintf("Received get metadata response: %s", metaDataRes.String()))
+	return nil
+}
+
+// IMPROVE: Remove this once we have a proper logging system.
```

Comment:
```suggestion
// IMPROVE(#420): Remove this once we have a proper logging system.
```

Not that @Gustavobelfort is really close to finishing #420, so I think we'll need to do it in this PR.

Reply:
noted, thanks.